### PR TITLE
build as dkms

### PR DESCRIPTION
**Pull request by vicamo**
Header files in `backport-include` was copied from each kernel release and merged. For those who would like to copy the source files into kernel tree and compile inlinely, it can be skipped. Debianization is mostly ready in https://code.launchpad.net/~vicamo/+git/intel-vsc-dkms/+ref/ubuntu/devel .

### CODE REVIEW

File: drivers/misc/mei/Makefile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (c) 2022 Canonical Ltd.
+
+obj-$(CONFIG_INTEL_MEI_VSC) += mei-vsc.o
+mei-vsc-objs := spi-vsc.o
+mei-vsc-objs += hw-vsc.o
+
+ccflags-y += -I$(src)/../../../backport-include/drivers/misc/mei
```
Comment (vicamo) [inline drivers/misc/mei/Makefile:8]: It should be nice to add a comment above this line saying it should be skipped when compiled in tree.

**Comment by wentongwu** [comment]
the added Makefile in sub folder(like gpio, i2c, etc) will block other real hardware building on ADL if vendors still choose in-tree building. Please delete these Makefiles except the root one. And add the objs and including in the root Makefile. Thanks

**Comment by wentongwu** [comment]
the added Makefile in sub folder(like gpio, i2c, etc) will block other real hardware building on ADL if vendors still choose in-tree building. Please delete these Makefiles except the root one. And add the objs and including in the root Makefile. Thanks

**Review by wentongwu — CHANGES_REQUESTED** [review]
the added Makefile in sub folder(like gpio, i2c, etc) will block other real hardware building on ADL if vendors still choose in-tree building. Please delete these Makefiles except the root one. And add the objs and including in the root Makefile. Thanks

**Comment by wentongwu** [comment]
@vicamo why you close this PR? it just need a little change to merge it.

**Pull request closed without merging**